src/main/java/org/jitsi/jigasi/lobby/Lobby.java

```diff
@@ -223,23 +224,56 @@ public void invitationReceived(ChatRoomInvitationReceivedEvent chatRoomInvitatio
     }
 
     /**
-     * Participant is kicked if rejected on join.
+     * Participant is kicked if rejected on join. This method handles the lobby rejection and lobby room destruction.
+     * Participant receives LOCAL_USER_LEFT if lobby is disabled.
      *
      * @param localUserChatRoomPresenceChangeEvent <tt>LocalUserChatRoomPresenceChangeEvent</tt> contains reason.
      */
     @Override
     public void localUserPresenceChanged(LocalUserChatRoomPresenceChangeEvent localUserChatRoomPresenceChangeEvent)
     {
-        if (localUserChatRoomPresenceChangeEvent.getEventType() == LocalUserChatRoomPresenceChangeEvent.LOCAL_USER_KICKED)
+        try
         {
-            if (this.lobbyListeners != null)
+            if (localUserChatRoomPresenceChangeEvent.getChatRoom().equals(this.mucRoom))
             {
-                for (LobbyListener ll : this.lobbyListeners)
+                if (localUserChatRoomPresenceChangeEvent.getEventType() == LocalUserChatRoomPresenceChangeEvent.LOCAL_USER_KICKED)
                 {
-                    ll.onLobbyAccessRejected(this);
+                    if (this.lobbyListeners != null)
+                    {
+                        for (LobbyListener ll : this.lobbyListeners)
+                        {
+                            ll.onLobbyAccessRejected(this);
+                        }
+                    }
+                }
+
+                if (localUserChatRoomPresenceChangeEvent.getEventType() == localUserChatRoomPresenceChangeEvent.LOCAL_USER_LEFT)
+                {
+                    String alternateAddress = localUserChatRoomPresenceChangeEvent.getAlternateAddress();
+
+                    Jid mainRoomJid = null;
+
+                    if (alternateAddress != null)
+                    {
+                        String mainRoomIdentifier = alternateAddress;
```

Comment:
You don't need this extra variable, right?

Reply:
Removed.